scvid/callbacks/distributed_pca.py

```diff
@@ -0,0 +1,90 @@
+# Copyright Contributors to the Cellarium project.
+# SPDX-License-Identifier: BSD-3-Clause
+
+import math
+
+import lightning.pytorch as pl
+import torch
+import torch.distributed as dist
+
+from scvid.module import IncrementalPCA
+
+
+class DistributedPCA(pl.Callback):
+    """
+    Distributed PCA.
```

Comment:
Please write a good amount of documentation about the logic, because it is beautiful! :)

Reply:
... and the code is very compact and efficient!  you're crazy.